### PR TITLE
Improve user logging file structure

### PR DESCRIPTION
**Pull request by ns1123**
Modified the user logging directory to prevent errors from occurring when the user running a command is different from the alluxio installation user.

@ZacBlanco or @aaudiber PTAL

### CODE REVIEW

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go [hunk 1]
```diff
@@ -305,10 +305,11 @@ func generateTarball(hadoopClients []string) error {
 	fmt.Printf("Creating %s:\n", tarball)
 
 	for _, dir := range []string{
-		"assembly", "client", "logs", "integration/fuse", "integration/checker",
+		"assembly", "client", "logs", "integration/fuse", "integration/checker", "logs/user",
 	} {
 		mkdir(filepath.Join(dstPath, dir))
 	}
+    os.Chmod(filepath.Join(dstPath, "logs/user"), 0777)
```
Comment (ZacBlanco) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go]: Don't forget to handle the error if we can't chmod the logs/user folder

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 2]
```diff
@@ -330,6 +330,13 @@ public String toString() {
           .setIgnoredSiteProperty(true)
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .build();
+  public static final PropertyKey USER_LOGS_DIR =
+      new Builder(Name.USER_LOGS_DIR)
+          .setDefaultValue(String.format("${%s}/user", Name.LOGS_DIR))
+          .setDescription("The path to store user log files.")
```
Comment (ZacBlanco) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java]: nit: I think someone reading this without background might not know what "user log files" are. Could we say something similar to the tune of "logs from alluxio client operations"? Even I'm not a huge fan of my suggestion, but I think I just want to make it more clear it's logs from client operations.

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 3]
```diff
@@ -3628,6 +3635,7 @@ private static String javadocLink(String fullyQualifiedClassname) {
     public static final String INTEGRATION_YARN_WORKERS_PER_HOST_MAX =
         "alluxio.integration.yarn.workers.per.host.max";
     public static final String LOGGER_TYPE = "alluxio.logger.type";
+    public static final String USER_LOGS_DIR = "alluxio.user.logs.dir";
```
Comment (ZacBlanco) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java]: nit: I think we try to keep these in alphabetical order

File: core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 2 above)
Comment (aaudiber) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: +1 for clarifying what a user log contains. How about "logs from alluxio CLI operations"?

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go [hunk 4]
```diff
@@ -334,7 +335,7 @@ func generateTarball(hadoopClients []string) error {
 	addAdditionalFiles(srcPath, dstPath, hadoopVersion, version)
 
 	chdir(cwd)
-	run("creating the distribution tarball", "tar", "-czvf", tarball, dstDir)
+	run("creating the distribution tarball", "tar", "-czpvf", tarball, dstDir)
```
Comment (aaudiber) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go]: Does this `-p` do anything? My understanding is that permissions are preserved by default on the create path, and `-p` only affects the extract path

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go [hunk 5]
```diff
@@ -305,11 +305,16 @@ func generateTarball(hadoopClients []string) error {
 	fmt.Printf("Creating %s:\n", tarball)
 
 	for _, dir := range []string{
-		"assembly", "client", "logs", "integration/fuse", "integration/checker",
+		"assembly", "client", "logs", "integration/fuse", "integration/checker", "logs/user",
 	} {
 		mkdir(filepath.Join(dstPath, dir))
 	}
 
+    err := os.Chmod(filepath.Join(dstPath, "logs/user"), 0777)
```
Comment (aaudiber) [inline dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go]: the Gopher idiom for this common pattern is
```go
if err := os.Chmod(filepath.Join(dstPath, "logs/user"), 0777); err != nil {
    return err
}

File: core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 2 above)
Comment (ns1123) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: Fixed.

File: core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 3 above)
Comment (ns1123) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: done.

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go (hunk 4 above)
Comment (ns1123) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go]: left in from me figuring out the same thing :). Fixed.

File: dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go (hunk 5 above)
Comment (ns1123) [reply dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go]: Noted and fixed.

**Comment by ns1123** [comment]
@aaudiber fixes done. Please help to merge.

**Comment by aaudiber** [comment]
@ns1123 just waiting for build to pass, then I'll merge

**Comment by aaudiber** [comment]
alluxio-bot, merge this please

**Comment by aaudiber** [comment]
alluxio-bot, cherry-pick to branch-1.8 please

**Pull request merged**